src/main/java/org/elasticsearch/index/query/BaseTermQueryBuilder.java

```diff
@@ -218,16 +212,16 @@ public boolean equals(Object obj) {
                Objects.equals(queryName, other.queryName);
     }
 
-    /** Read the given parameters. */
     @Override
-    public void readFrom(StreamInput in) throws IOException {
-        fieldName = in.readString();
-        value = in.readGenericValue();
-        boost = in.readFloat();
-        queryName = in.readOptionalString();
+    public QB readFrom(StreamInput in) throws IOException {
+        QB emptyBuilder = createBuilder(in.readString(), in.readGenericValue());
+        emptyBuilder.boost(in.readFloat());
```

Comment:
Here it should not matter if setters are used for boost and queryname. This was more like a general question about the pros / cons of using setters in readFrom. I'm fine with any approach, just wanted to know if it is bad practice to directly write to fields here.